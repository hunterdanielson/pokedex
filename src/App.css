.app-container {
  background-image: linear-gradient(to bottom right, #5e0000, #c00d0d);
  height: 1000px;
  color: white;
  text-shadow: 1px 1px black;
}

.selector-container {
  padding: 25px;
  display: flex;
  justify-content: space-around;
}

.pokemon-list-container {
  background-color: #8b0000;
}

.pokemon-data-container {
  display: grid;
  grid-template-columns: 25% 25% 25% 25%;
}

.full-pokemon-container {
  background-color: #c00d0d;
  margin: 5px;
  border: 2px solid black;
  border-radius: 2%;
  box-shadow: 5px 3px #5e0000;
  text-transform: capitalize;
}

.search-button {
  transform: scale(1.5);
  background: radial-gradient(#05fbfb, #29abe3);
  color: white;
  text-shadow: .5px .5px black;
  padding: 2px 15px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
}

img {
  background-color: white;
  border: solid black 2px;
  border-radius: 15%;
  height: 230px;
  width: 200px;
}

a:link, a:visited {
  background: radial-gradient(#05fbfb, #29abe3);
  color: white;
  text-shadow: .5px .5px black;
  padding: 2px 15px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
}

a:hover, a:active {
  background: radial-gradient(#29abe3, #003300);
}