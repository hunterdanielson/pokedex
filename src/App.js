import React, { Component } from 'react'
import request from 'superagent';
import PokemonList from './PokemonList.js';

const data = [{}]

export default class App extends Component {
  state = { typeQuery: null, searchQuery: null, data: data }
  
  handleType = (e) => {
    this.setState({ typeQuery: e.target.value })
  }

  handleChange = (e) => {
    this.setState({ searchQuery: e.target.value })
  }

  handleSearch = async () => {
    const fetchedData = await request.get(`https://alchemy-pokedex.herokuapp.com/api/pokedex?${this.state.typeQuery}=${this.state.searchQuery}`)
    this.setState({ data: fetchedData.body.results })
  }
  render() {
    return (
      <div>
        <select onChange={this.handleType}>
          <option value=""></option>
          <option value="pokemon">Pokemon Name</option>
          <option value="type">Type</option>
          <option value="ability">Ability</option>
          <option value="attack">Attack</option>
        </select>
        <input onChange={this.handleChange}></input>
        
        <button onClick={this.handleSearch}>Search</button>
        <PokemonList pokemons={this.state.data} />
      </div>
    )
  }
}
