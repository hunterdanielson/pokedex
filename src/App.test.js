import React from 'react';
import { shallow } from 'enzyme';
import App from './App.js';

test('testing App', () => {
    const wrapper = shallow(<App />)

    expect(wrapper).toMatchSnapshot();
})