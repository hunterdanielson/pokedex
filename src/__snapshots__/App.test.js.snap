// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing App 1`] = `ShallowWrapper {}`;
